extension/memorylimiterextension/memorylimiter_test.go

```diff
@@ -95,3 +96,8 @@ func TestMemoryPressureResponse(t *testing.T) {
 		})
 	}
 }
+
+func TestMiddleware(t *testing.T) {
```

Comment:
Instead of having a test for this, could we use the type checker to assert interface implementation?

You could put the following above line 19 in `memorylimiter.go` for example:

```golang
var _ extensionmiddleware.GRPCServer = (*memoryLimiterExtension)(nil)
var _ extensionmiddleware.HTTPServer = (*memoryLimiterExtension)(nil)
```